operators/pkg/controller/elasticsearch/nodespec/resources.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package nodespec
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	commonv1alpha1 "github.com/elastic/cloud-on-k8s/operators/pkg/apis/common/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/keystore"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/label"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/settings"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/elasticsearch/sset"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+)
+
+// Resources contain per-NodeSpec resources to be created.
+type Resources struct {
+	StatefulSet     appsv1.StatefulSet
+	HeadlessService corev1.Service
+	Config          settings.CanonicalConfig
+}
+
+type ResourcesList []Resources
+
+func (l ResourcesList) StatefulSets() sset.StatefulSetList {
+	ssetList := make(sset.StatefulSetList, 0, len(l))
+	for _, resource := range l {
+		ssetList = append(ssetList, resource.StatefulSet)
+	}
+	return ssetList
+}
+
+func BuildExpectedResources(es v1alpha1.Elasticsearch, keystoreResources *keystore.Resources) (ResourcesList, error) {
+	nodesResources := make(ResourcesList, 0, len(es.Spec.Nodes))
+
+	for _, nodeSpec := range es.Spec.Nodes {
```

Comment:
I think so because it looks good in the yaml manifest?

```yaml
spec:
  version: 7.2.0
  nodes:
  - name: hot
    nodeCount: 3
  - name: warm
    nodeCount: 3
```

Reply:
I do agree nodes look nice, but I have mixed feelings about how well it describes what it is:) My concern is that it's a set of sets of nodes and not a node list as one might assume. I initially thought we changed it recently, but it seems it's been there a while - let's leave it as it is, unless more people share my concern.

Reply:
I get your point.
Do you have a suggestion for a better name? Maybe `nodeSpecs`, `nodeTemplates`?

Reply:
nodeGroups? nodeSets?

Reply:
> @anyasabo I think the samples close to the code (config/samples) are updated already. Are you referring to examples in our documentation?

Exactly

Reply:
Let's discuss it with the team (outside the scope of this PR).

Reply:
Naming is hard. It reminds me of the day we went from `topologies` to `topology`: #188, then from `topology` to `nodes`: 6b73b463.
`nodeGroups` or `nodesGroups`? may be more explicit but more verbose. 
So, I'm fine with `nodes`.